assets/src/features/room-page/utils/usePinning.tsx

```diff
@@ -0,0 +1,46 @@
+import { useState } from "react";
+
+export type PinningApi = {
+  pinnedTileIds: string[];
+  pin: (tileId: string) => void;
+  unpin: (tileId: string) => void;
+  pinIfNotAlreadyPinned: (tileId: string) => void;
+};
+
+const usePinning = (): PinningApi => {
+  const [pinnedTileIds, setPinnedTileIds] = useState<string[]>([]);
+  const [pinnedTileIdHistory, setPinnedTileIdHistory] = useState<string[]>([]);
+  const historyLimit = 20;
+
+  const pin = (newTileId: string) => {
+    const updateHistory = (tileId: string) => {
```

Comment:
when you change an array to a set it will be more accurate to call it `addToHistory`